lib/jormungandr/src/Cardano/Wallet/Jormungandr/Api.hs

```diff
@@ -102,7 +102,7 @@ type PostSignedTx
 -- TODO: Replace SignedTx with something real
 data SignedTx
 
-newtype BlockId = BlockId (Hash "block")
+newtype BlockId = BlockId (Hash "BlockHeader")
```

Comment:
Yep, I think `Hash "BlockHeader"` is correct, and `BlockId` probably isn't needed - as either a newtype or type synonym.

Reply:
It doubles as an `ApiT` to avoid orphans so it is needed. I kept it in `JormungandrLayer` for now.

Also I introduced it because jormungandr/`JormungandrLayer` often refers to "id" in the method-names.